38 settings screen - theme change and export/import

EEtele opened the pull request:
New screen: Settings
-Option to change color theme
-Export tasks to json file
-Import tasks from json file

EEtele committed:
Prototype(pre alpha tier) theme selector

EEtele committed:
tiny bugfix

EEtele committed:
NewNoteScreen changes

EEtele committed:
Changed minSdk, lowered to 24
Had to delete 2 .xml icons for it to be possible

EEtele committed:
Merge branch 'master' of https://github.com/EmmChriss/skuld-android into 38-settings-screen

 Conflicts:
	app/src/main/java/com/lab/skuld/ui/App.kt

EEtele committed:
Export and Import functionality based on JSON and GSON

EmmChriss commented on app/src/main/java/com/lab/skuld/MainActivity.kt:
```diff
@@ -5,10 +5,33 @@ import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import com.lab.skuld.ui.App
 
+
```
Don't leave unused comments lying around

EEtele replied:
:(((

EmmChriss commented on app/src/main/java/com/lab/skuld/ui/screens/NewNoteScreen.kt:
```diff
@@ -143,11 +156,18 @@ fun ShowNewNoteScreen(documentt: Event = emptyEventorTask){
 
 
-
+    val focusManager = LocalFocusManager.current
```
This should be removed, pull from master and test if App.kt:208 implementation of this handles your use-case, but I suspect it does

EEtele replied:
Removed, your implementation handles this use-case too.

EmmChriss commented on app/src/main/java/com/lab/skuld/ui/screens/SettingsScreen.kt:
```diff
@@ -0,0 +1,210 @@
+    }
+}
+/*
+///////////////////////First attempt, read at own risk////////////////////////
```
xd

EmmChriss commented on app/src/main/java/com/lab/skuld/ui/theme/Theme.kt:
```diff
@@ -47,15 +41,18 @@ private val LightColorPalette = lightColors(
 @Composable
 fun SkuldFrontendTheme(
+
     darkTheme: Boolean = isSystemInDarkTheme(),
```
Hmm, but then we aren't testing for system theme. It would be nice to have a "System theme" option that does what we were doing previously

EEtele replied:
Added(and tested) new default option "System theme" to settings.

EEtele committed:
-removed some unused comments
-removed unnecesary focus management from NewNoteScreen.kt
-added "System theme" default option for color theme in settings

EmmChriss commented on app/src/main/java/com/lab/skuld/ui/screens/NewNoteScreen.kt:
```diff
@@ -117,6 +120,13 @@ val emptyEventorTask = Event(id = "", startDate = Date(), endDate = Date(), titl
+
+
+
+
```
my brother in christ
